api/src/database/helpers/schema/types.ts

```diff
@@ -19,37 +21,14 @@ export abstract class SchemaHelper extends DatabaseHelper {
 		});
 	}
 
-	async changeToText(
-		table: string,
-		column: string,
-		options: { nullable?: boolean; default?: any } = {}
-	): Promise<void> {
-		await this.knex.schema.alterTable(table, (builder) => {
-			const b = builder.string(column);
```

Comment:
The `changeToText` uses `.string()` for all vendors except for `oracle` and `crdb`. It was necessary in order to support these two vendors. Ref: https://github.com/directus/directus/pull/5462 & https://github.com/directus/directus/pull/10113.

After this PR, existing fields such as `url` and `collections` in `directus_webhooks` will be set to `text` type. Whereas the existing installations would still have them as `string` type.

@rijkvanzanten How should we approach this issue?
- Should we add database specific migrations so we will be able to run `changeToType(...'string')` for all vendors except for `oracle` and `crdb` which will run `changeToType(...'text')` to maintain the existing behaviour?
- Or should we standardise it where we add a migration so that all databases will have the same `text` type from now onwards? 

The latter option would definitely be less efficient, database wise.

Reply:
> Or should we standardise it where we add a migration so that all databases will have the same text type from now onwards?

Even though it's less efficient on most databases, I'm worried about having databases in different system states based on their vendor. 🤔 For example if we rely on the database for max length, it's gonna work differently based on what vendor you're using

Reply:
@rijkvanzanten The fields are of type `string` with a length of `255`, which is the default value when length is not provided.  
An example is the `collections` field in `directus_webhooks`. https://github.com/directus/directus/blob/71e23a024eeefd4ebd691ae6a2a2c255f84f98e6/api/src/database/seeds/14-webhooks.yaml#L32-L35
How should we proceed?

Reply:
Difficult difficult question. Before we set that limit to 64 as that was the table length limit on Postgres, but that was changed to 255 to account for sqlite names if I recall correctly 🤔 Making those text doesn't make too much sense as we know they're smaller values. That being said, is it that noticeable of a performance difference nowadays? There's whole ORM abstractions that rely exclusively on storing everything unstructured

Reply:
I believe the main concern is the slight difference of field type across different Directus instances depending on their installation "period". Hopefully this table helps to summarize it:

|                                   | CockroachDB | Oracle | Other DBs |
| --------------------------------- | ----------- | ------ | --------- |
| New installations in the past     | -           | CLOB   | text      |
| New installations after PR #10113 | text        | text   | string    |
| New installations after this PR   | text        | text   | text      |

As there's that slight difference, there'll be a form of "desync" between several instances. Another potential issue could be with schema migrations (as the schema file will indicate them differently) across Directus instances that were initialized with "different" webhooks migrations. 🤔

With all that being said, I do also want to acknowledge that as `changeToText` helper was exclusively used in `directus_webhooks` migrations only, and since webhooks will be deprecated in favor of Flows in the future, this whole concern will technically be obsolete when that happens 😅

Reply:
We can also make use of the `isOneOfClients` helper:

https://github.com/directus/directus/blob/52598538f047e093e6629cfe3680eddbbd7f5f19/api/src/database/migrations/20210626A-change-filesize-bigint.ts#L7-L9

and wrap it as an IF-ELSE for `changeToType` in webhook migrations to keep the existing "text for crdb and oracle, string for others" behavior 👍